### PR TITLE
Fix inclusion of prereleases

### PR DESCRIPTION
**Pull request by marcodejongh**
My original implementation of the include prereleases fix
was too naive,as it just forced yarn to include prereleases everywhere.
Which resulted in yarn add also resolving to prereleases
yarn add package@^1.0.0
Would resolve to
package@1.0.1-alpha
if that existed. This is obviously undesirable, but we still want to dedupe
across alpha's, i.e:
"@atlaskit/avatar-group@8.4.0-alpha-0371f4c06e53", "@atlaskit/avatar-group@^8.0.0"
lock entries.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
     "request": "^2.87.0",
     "request-capture-har": "^1.2.2",
     "rimraf": "^2.5.0",
-    "semver": "atlassian-forks/node-semver.git#refs/heads/alwasIncludePrerelease",
```
Comment (lukebatchelor) [inline package.json:46]: Why was this changed sorry?

**Comment by lukebatchelor** [comment]
I'm having trouble understanding what the actual change was here.

We moved away from the forked `semver`, and then passed the `includePrerelease` flag to it in a bunch of places, but I'm not sure how that's connected to `yarn add` using the alpha versions too. I guess I'm looking for some code that was either reversing the previous changes or something that's obviously checking what command we're in for example?

Does that question make sense?

**Review by marcins — APPROVED** [review]

**Review by ReDrUm — APPROVED** [review]

**Comment by marcodejongh** [comment]
@lbatch Yeah, your question makes sense. For the original implementation, I just switched yarn to a version of semver that always included prereleases all the time. This actually worked fine 99%, I've been testing this patch locally and the only way I could reproduce the bug was with:

```
yarn add @atlaskit/atlassian-switcher@^13.0.5
```

Which would end up using `13.0.6-alpha-xxx` in the lock file (but putting 13.0.5 in the package.json)
So my original theory was that it was because of there being a newer version than 13.0.5 being available.
Meaning that other packages where this is true would also be subject to the same bug. I searched or registry and found that
Button is currently at 16.0.0 but there's 16.1.0-alpha's in the registry. But installing 16.0.0:

```
yarn add @atlaskit/button@^16.0.0
```

Did result in the product using 16.0.0 and not the 16.1.0-alpha like I would have expected.

So this bug thankfully was a lot less widespread than I had originally feared. I do wonder whats special about atlassian-switcher.

With the changes in this PR we only include prereleases when checking the yarn.lock file for correctness.
Without this fix the following lock entry is invalid according to yarn:

```
"@atlaskit/atlassian-switcher@^13.1.0-alpha" "@atlaskit/atlassian-switcher@^13.0.5":
 version "13.1.0-alpha"
```

So it would throw an error and instead fix the yarn lock to:


```
"@atlaskit/atlassian-switcher@^13.1.0-alpha":
 version "13.1.0-alpha"


"@atlaskit/atlassian-switcher@^13.0.5":
 version "13.0.5"
```

This is problematic because it makes deduplication impossible. This new fix _only_ includes prereleases when evaluating lockfile correctness, it doesn't change any of the semver evaluation behaviour anywhere else inside yarn.

File: package.json (hunk 1 above)
Comment (marcodejongh) [reply package.json:46]: See: https://github.com/atlassian-forks/yarn/pull/6#issuecomment-895010465
